http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -633,17 +633,27 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
       // Create the cluster synchronously
       createClusterResultOpt <- msg.cluster
         .traverse { c =>
-          gkeInterp.createCluster(CreateClusterParams(c.clusterId, msg.project, List(c.defaultNodepoolId), false))
-        }
-        .adaptError {
-          case e =>
-            PubsubKubernetesError(
-              AppError(e.getMessage, ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Cluster, None),
-              Some(msg.appId),
-              false,
-              msg.cluster.map(_.defaultNodepoolId),
-              msg.cluster.map(_.clusterId)
-            )
+          gkeInterp
+            .createCluster(CreateClusterParams(c.clusterId, msg.project, List(c.defaultNodepoolId), false))
+            .onError {
+              case _ =>
+                for {
+                  _ <- gkeInterp.deleteAndPollCluster(DeleteClusterParams(c.clusterId, msg.project))
```

Comment:
Does this do the deletion + polling synchronous to the pubsub request?

(I think that's fine, just curious)

Reply:
Yes. it seemed a bit heavy to make an async task when this portion is sync already. It possible that this could trigger message resending from google, but It's set to 5 mins and this is usually sufficient to delete the cluster.